File: Books Price Python/controller/sale_controller.py
from pymongo.collection import Collection
from model.sale import Sale
from utils.mongodb_connection import MongoDBConnection


class SaleController:
    def __init__(self):
        self.collection: Collection = (
            MongoDBConnection
            .get_database()
            .get_collection("Sales")
        )

    def save_sale(self, sale: Sale):
        document = {
            "product": sale.get_product(),
            "price": sale.get_price(),
            "quantity": sale.get_quantity(),
            "subtotal": sale.get_subtotal(),
            "iva": sale.get_iva(),
            "total": sale.get_total(),
            "date": sale.get_date()
        }

        self.collection.insert_one(document)

    def get_all_sales(self):
        sales = []

        for doc in self.collection.find():
            sale = Sale()
            sale.set_product(doc.get("product"))
            sale.set_price(doc.get("price"))
            sale.set_quantity(doc.get("quantity"))

            sales.append(sale)

        return sales

File: SaleSystem/src/main/java/ec/edu/espe/salesystem/controller/SaleController.java
package ec.edu.espe.salesystem.controller;

import com.mongodb.client.MongoCollection;
import ec.edu.espe.salesystem.model.Sale;
import ec.edu.espe.salesystem.utils.MongoDBConnection;
import java.util.ArrayList;
import java.util.List;
import org.bson.Document;
/**
 *
 * @author Arelis Samantha Bonilla Cruz, Student, @ESPE
 */
public class SaleController {
    private MongoCollection<Document> collection;

    public SaleController() {
        collection = MongoDBConnection
                .getDatabase()
                .getCollection("Sales");
    }

    public void saveSale(Sale sale) {
        Document document = new Document("product", sale.getProduct())
                .append("price", sale.getPrice())
                .append("quantity", sale.getQuantity())
                .append("subtotal", sale.getSubtotal())
                .append("iva", sale.getIva())
                .append("total", sale.getTotal())
                .append("date", sale.getDate());

        collection.insertOne(document);
    }

    public List<Sale> getAllSales() {
        List<Sale> sales = new ArrayList<>();

        for (Document doc : collection.find()) {
            Sale sale = new Sale();
            sale.setProduct(doc.getString("product"));
            sale.setPrice(doc.getDouble("price"));
            sale.setQuantity(doc.getInteger("quantity"));

            sales.add(sale);
        }
        return sales;
    }
}
